### PR TITLE
feat(python): bindings working

### PR DESCRIPTION
**Pull request by blu3beri**
Fixed the bindings from credx to anoncreds and updated the api.

Still some issues with the tests, mostly regarding revocation, as that is not yet finished from the ffi side.

Work funded by the Government of Ontario.

Signed-off-by: blu3beri <blu3beri@proton.me>

### CODE REVIEW

**Review by TimoGlastra — APPROVED** [review]
LGTM

**Comment by TimoGlastra** [comment]
Should we somehow enable the tests for the python wrapper? Was that disabled in the move from the indy-shared-rs repo?

File: wrappers/python/anoncreds/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
-"""Indy-Credx Python wrapper library"""
+"""Anoncreds Python wrapper library"""
```
[the remark this answers is not in the document]
Comment (blu3beri) [reply wrappers/python/anoncreds/__init__.py:1]: Yeah I also mentioned it with the javascript wrapper. My preference goes to Anoncreds as more of a name and less of an abbreviation. I am perfectly fine with switching but I think we need a definitive answer. 

@swcurran Is this something worthwhile discussing during the WG or do you have a very strong preference?

File: anoncreds/src/ffi/cred_req.rs [hunk 2]
```diff
@@ -25,18 +25,20 @@ pub extern "C" fn anoncreds_create_credential_request(
...
+
+        let prover_did = prover_did.as_opt_str();
+
         // Here we check whether the identifiers inside the cred_def (schema_id, and issuer_id)
         // are legacy or new. If they are new, it is not allowed to supply a `prover_did` and a
         // random string will be chosen for you.
-        let cred_def = cred_def.load()?;
-        let cred_def: &CredentialDefinition = cred_def.cast_ref()?;
-        if cred_def.schema_id.is_uri() || cred_def.issuer_id.is_uri() {
+        if (cred_def.schema_id.is_uri() || cred_def.issuer_id.is_uri()) && prover_did.is_some() {
```
[the remark this answers is not in the document]
Comment (blu3beri) [reply anoncreds/src/ffi/cred_req.rs:36]: Yeah good point actually. There is, I believe, an `is_legacy_identifier` function. Just a bit curious now how we can make the `is_uri` or `is_new_identifier` more strict.

**Comment by blu3beri** [comment]
> Should we somehow enable the tests for the python wrapper? Was that disabled in the move from the indy-shared-rs repo?

Yes they should be enabled. I assumed they were. Will also add that in this PR.

File: anoncreds/src/ffi/cred_req.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply anoncreds/src/ffi/cred_req.rs:36]: Don't think we need to. We should just call it is_uri i think and if it validates it's fine.

**Comment by TimoGlastra** [comment]
> > Should we somehow enable the tests for the python wrapper? Was that disabled in the move from the indy-shared-rs repo?
> 
> 
> 
> Yes they should be enabled. I assumed they were. Will also add that in this PR.

Could be they're disbaled when forking. Took me a while to figure out last time that they can be disbaled frim settings

**Pull request merged**